main.py

```diff
@@ -6,106 +6,277 @@
-        elif shape.method == 'polygon':
-            pygame.draw.polygon(SCREEN, shape.colour, position)
-        elif shape.method == 'line':
-            pygame.draw.line(SCREEN, shape.colour, position[0], position[1], shape.radius)
-
-    target_shape = shapes[random.randint(0, 3)]
-    while target_shape.method == last_shape:
         target_shape = shapes[random.randint(0, 3)]
+        while target_shape.method == last_shape:
+            target_shape = shapes[random.randint(0, 3)]
 
-    font = pygame.font.Font(None, 48)
-    text = font.render(f"{current_round}. Please select the {target_shape.type}", True, (255, 255, 255))
-    text_rect = text.get_rect(center=(400, 50))
-    SCREEN.blit(text, text_rect)
+    heading = f'{current_round}. Please select the {target_shape.type}'
+    drawing.draw_heading(heading)
 
     return target_shape
 
 
+def draw_third_stage(drawing: Drawing) -> Shape:
+    """Draw the next round screen and return the chosen shape"""
+    drawing.clear_screen()
+    random.shuffle(IAPS)
+    figure = IAPS.pop()
+    plus = get_iap(figure)
+    image = pygame.image.load(figure)
+    SCREEN.blit(image, (0, 0))
+    return plus
+
+
+def draw_whitespace(drawing: Drawing):
+    """Draw the next round screen and return the chosen shape"""
+    drawing.clear_screen((255, 255, 255))
+
+
 shapes = load_shapes()
 positions = ["top_left", "top_right", "bottom_left", "bottom_right"]
+subject_ID = input('Insert Subject ID: \n')
+today = date.today()
+day = today.strftime("%d/%m/%Y")
+subject = Subject(subject_ID, day, {})
+subject.create_data_template()
+Round = Round(['Game', 'SingleChoiceGame', 'Stimulation'], 1, '0', 81, 'Game')
+SimpleRecording = SimpleRecording()
+Instance = Instance(subject, SimpleRecording, Round)
+
+start = 0
+while start == 0:
+    start = input('Ready to start the window? (1 - yes, 0 - no): \n')
 
 pygame.init()
-SCREEN = pygame.display.set_mode((800, 600))
+SCREEN = pygame.display.set_mode((1920, 1080))
+
+position_right = 0
+while position_right == 0:
+    position_right = input('Is the window correctly positioned? (1 - yes, 0 - no): \n')
 
-# TODO: Build structure / classes to handle events better
 generate_new_round = pygame.USEREVENT + 1
 pygame.time.set_timer(generate_new_round, 5000)
 
-max_rounds = 24
+max_rounds = 81
 current_round = 1
-
+whitespace_rounds = [25, 26, 28, 29, 31, 32, 34, 35, 37, 38, 40, 41, 43, 44, 46, 47, 49, 50, 52, 53, 55, 56, 58, 59, 61,
+                     62, 64, 65, 67, 68, 70, 71, 73, 74, 76, 77, 79, 80]
+drawing = Drawing()
 last_shape = None
-winning_shape = draw()
+winning_shape = draw(drawing)
 clickable = True
+stage = 'Game'
+Instance.add_timestamp(current_round, stage)
 
 while True:
     for events in pygame.event.get():
         if events.type == pygame.QUIT:
             pygame.quit()
             sys.exit()
 
-        if events.type == generate_new_round and current_round <= 24:
+        if events.type == generate_new_round and current_round <= 81:
```

Comment:
We built a class to represent a round and an instance class to handle logic with incrementation and switching stage etc, somehow we've gone back to the 81 round count iterator and tracking rounds separately here.

It is a bit of a mess and hard to follow, but since you need to run this experiment for real tomorrow and it's now working for you, just leave it as it is. (For now)

You should also consider using elif for the current round checks here.

Reply:
If we're going to use this later on, we need to remake the round logic from the ground up. The subject save function was also broken, but I changed it to work for my own devices. The problem was syntax for the json class, I believe

Reply:
I am planning on working on this again on Thursday as well, when I don't have a time limit looming over me, cause I know a lot of that could be better, but I took 8 hours to make this one work, so I need a wee break XD

Reply:
Absolutely, take all the time you need, you don't owe me anything I'm the one assisting you with your project / learning. 😅 This does seem like a good project for you to come back to and collaborate on to improve your skills so I'd be glad to help you whip this into better shape and slowly introduce better design patterns to replace the old stuff.